doc/source/dev/index.rst

```diff
@@ -2,15 +2,331 @@
+
+Divergence between ``upstream/master`` and your feature branch
+--------------------------------------------------------------
+
+If GitHub indicates that the branch of your Pull Request can no longer
+be merged automatically, you have to incorporate changes that have been made
+since you started into your branch. Our recommended way to do this is to
+`rebase on master <rebasing-on-master>`.
+
+Guidelines
+----------
+
+* All code should have tests (see `test coverage`_ below for more details).
+* All code should be `documented <docstring-standard>`.
+* No changes are ever committed without review and approval by a core
+  team member.  Ask on the `mailing list`_ if you get no response to your pull
+  request.
+
+Stylistic Guidelines
+--------------------
+
+* Set up your editor to follow `PEP08 <https://www.python.org/dev/peps/
+  pep-0008/>`_ (remove trailing white space, no tabs, etc.).  Check code with
+  pyflakes / flake8.
+
+* Use numpy data types instead of strings (``np.uint8`` instead of
+  ``"uint8"``).
+
+* Use the following import conventions::
+
+   import numpy as np
+
+   cimport numpy as cnp  # in Cython code
+
+* Use ``np_intp`` as data type for all indexing, shape and size variables
+  in C/C++ and Cython code.
+
+* For C code, see the `numpy-c-style-guide`
+
+
+Test coverage
+-------------
+
+Pull requests (PRs) that modify code should either have new tests, or modify existing
+tests to fail before the PR and pass afterwards. You should `run the tests
+<development-environment>` before pushing a PR.
+
+Tests for a module should ideally cover all code in that module,
+i.e., statement coverage should be at 100%.
+
+To measure the test coverage, install
+`pytest-cov <https://pytest-cov.readthedocs.io/en/latest/>`__
+and then run::
+
+  $ python runtests.py --coverage
+
+This will create a report in `build/coverage`, which can be viewed with
+
+```
+firefox build/coverage/index.html
+```
+
+Building docs
+-------------
+
+To build docs, run ``make`` from the ``doc`` directory. ``make help`` lists
+all targets. For example, to build the HTML documentation, you can run:
+
+.. code:: sh
+
+    make html
+
+Then, all the HTML files will be generated in ``doc/build/html/``.
+To rebuild a full clean documentation, run:
+
+.. code:: sh
+
+    make dist
+
+Requirements
+~~~~~~~~~~~~
+
+`Sphinx <http://www.sphinx-doc.org/en/stable/>`__ and LaTeX are needed to build
+the documentation.
+
+**Sphinx:**
+
+Sphinx and other python packages needed to build the documentation
+Python dependencies can be installed using:
+
+.. code:: sh
+
+    pip install -r sphinx matplotlib scipy
+
+**LaTeX Ubuntu:**
+
+.. code:: sh
+
+    sudo apt-get install -qq texlive texlive-latex-extra dvipng
+
+**LaTeX Mac:**
+
+Install the full `MacTex <https://www.tug.org/mactex/>`__ installation or
+install the smaller
+`BasicTex <https://www.tug.org/mactex/morepackages.html>`__ and add *ucs*
+and *dvipng* packages:
+
+.. code:: sh
+
+    sudo tlmgr install ucs dvipng
```

Comment:
this needs removing. if we keep the latex mention at all, it should just list package names or link to another page with more details. too many ways to do this, and too much of a hurdle

Reply:
removed